lib/client.js

```diff
@@ -154,22 +220,29 @@ module.exports = function (dependencies) {
         resolve({ device, error });
       });
 
+      if (logger.enabled) {
+        this.session.on("close", () => {
+          logger("Request closed");
+        });
+
+        this.session.on("frameError", (frameType, errorCode, streamId) => {
+          logger(`Request frame error: (frameType: ${frameType}, errorCode ${errorCode}, streamId: ${streamId})`);
+        });
+      }
+
       request.end();
     });
   };
 
   Client.prototype.shutdown = function shutdown(callback) {
+    if (logger.enabled) {
```

Comment:
nit: Could return early (and call callback?) if this.isDestroyed is already true (i.e. something called shutdown earlier)?

Out of scope for this PR, though, and not a concern for well written programs

Reply:
done